### PR TITLE
Working fork

### PR DESCRIPTION
**Pull request by andrioid**
Hi. I needed to import Pinboard posts to my blog, so I started using your plugin.

There was something missing, so I added it for my purposes. You can merge it, if you want. I can also adapt it, if you have any comments.

### CODE REVIEW

File: gatsby-node.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      resolve(data.feed);
+    });
+  });
+}
+
+const createChildren = (entries, parentId, createNode) => {
+  const childIds = [];
+  entries.forEach(entry => {
+    childIds.push(entry.link);
+    console.log('entry', entry);
```
Comment (jondubin) [inline gatsby-node.js]: ✂️

File: gatsby-node.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+};
+
+async function sourceNodes({ boundActionCreators }, { rssURL }) {
+  const { createNode } = boundActionCreators;
+  console.log('fetching rss feed');
+  const data = await promisifiedParseURL(rssURL);
+  if (!data) {
+    return;
+  }
+  // console.log('data from rss', data);
```
Comment (jondubin) [inline gatsby-node.js]: ✂️

File: gatsby-node.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  const { title, description, link, entries } = data;
+  const childrenIds = createChildren(entries, link, createNode);
+  const feedStory = {
+    id: link,
+    title,
+    description,
+    link,
+    // children: [],
```
Comment (jondubin) [inline gatsby-node.js]: ✂️

File: gatsby-node.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    parent: null,
+    children: childrenIds
+  };
+
+  feedStory.internal = { type: 'rssFeed', contentDigest: createContentDigest(feedStory) };
+
+  console.log('creating', feedStory);
```
Comment (jondubin) [inline gatsby-node.js]: ✂️

File: gatsby-node.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+  createNode(feedStory);
+
+  // Process data into nodes.
+  // data.forEach(datum => createNode(datum));
```
Comment (jondubin) [inline gatsby-node.js]: ✂️

File: package.json [hunk 6]
```diff
@@ -1,12 +1,16 @@
 {
-  "name": "gatsby-source-rss",
-  "version": "1.0.0",
+  "name": "gatsby-source-rss-fork",
```
Comment (jondubin) [inline package.json]: Wanna change this back to `gatsby-source-rss` so that it's the "official", unforked version?

File: src/gatsby-node.js [hunk 7]
```diff
@@ -1,21 +1,69 @@
...
-  { rssURL },
-) {
+async function sourceNodes({ boundActionCreators }, { rssURL }) {
   const { createNode } = boundActionCreators;
+  console.log('fetching rss feed');
   const data = await promisifiedParseURL(rssURL);
+  if (!data) {
+    return;
+  }
+  // console.log('data from rss', data);
```
Comment (jondubin) [inline src/gatsby-node.js]: ✂️

File: src/gatsby-node.js [hunk 8]
```diff
@@ -1,21 +1,69 @@
...
+  const { title, description, link, entries } = data;
+  const childrenIds = createChildren(entries, link, createNode);
+  const feedStory = {
+    id: link,
+    title,
+    description,
+    link,
+    // children: [],
```
Comment (jondubin) [inline src/gatsby-node.js]: ✂️

File: src/gatsby-node.js [hunk 9]
```diff
@@ -1,21 +1,69 @@
...
+    parent: null,
+    children: childrenIds,
+  };
+
+  feedStory.internal = { type: 'rssFeed', contentDigest: createContentDigest(feedStory) };
+
+  console.log('creating', feedStory);
```
Comment (jondubin) [inline src/gatsby-node.js]: ✂️

File: src/gatsby-node.js [hunk 10]
```diff
@@ -1,21 +1,69 @@
...
+
+  createNode(feedStory);
 
   // Process data into nodes.
-  data.forEach(datum => createNode(processDatum(datum)));
+  // data.forEach(datum => createNode(datum));
```
Comment (jondubin) [inline src/gatsby-node.js]: ✂️

**Review by jondubin — COMMENTED** [review]
Hey! Looks awesome. Thank you for taking the time to work on this. Just left some comments around removing dead code, but otherwise LGTM

File: package.json (hunk 6 above)
Comment (andrioid) [reply package.json]: Of course.

**Comment by andrioid** [comment]
I've removed the comments and the console.logs. So this should be in a semi-good shape. Let me know if there are other concerns.

**Comment by andrioid** [comment]
@jondubin Ready to merge and publish?

**Review by jondubin — APPROVED** [review]

**Comment by jondubin** [comment]
Yep, feel free to merge it in!

**Comment by andrioid** [comment]
I can't merge it. Github says "Only those with 'write access' to this repository can merge pull requests"

**Comment by andrioid** [comment]
Can you bump the version and publish to npm or give me access?

**Pull request merged**